### ISSUE TITLE
RPM packages fail to run under Fedora 34/35

### ISSUE THREAD

**Issue by jelabarre59** [moneymanagerex/moneymanagerex#4075]
MMEX version:
 - [x] 1.5.x
 - [ ] 1.6.x

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [ ] Mac OSX
 - [X] Linux 

# Description of the bug

MMEX fails immediately with "mmex: symbol lookup error: mmex: undefined symbol: _ZNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEC1EPKwRKS3_, version WXU_3.0".   Cannot run the application

Please describe your issue with details.
Add screenshot or other files if needed.

Install MMEX RPM under Fedora (tried with Fedora 34 & 35).  Start application from command line so you can see the error code (if started from an icon, nothing shows up at all).   Fails immediately with "symbol lookup error"

Tested with MMEX packages:
- mmex-1.3.6-1.fc30.x86_64.rpm
- mmex-1.5.0-1.fc30.x86_64.rpm
- mmex-1.5.3-1.fc30.x86_64.rpm
- mmex-1.5.4_Beta.1-1.fc30.x86_64.rpm
- mmex-1.6.0_Alpha.1-1.fc30.x86_64.rpm

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. download MMEX RPM package (any of the above)
2. install MMEX from RPM
3. run mmex from command line so you can see the failure message

…

Expected result:
Application starts and runs

Actual result:
Application fails immediately

# Additional information
The Github page suggests compiling the application rather than running the binary, but the compilation instructions are for MacOS, not Linux.

If you have a backtrace for a crash or a warning, paste it here.

**Comment by jota2016** [comment]
Hi jelabarre59,
When compiling, you only need to ignore the MacOS settings.
[Here](https://github.com/moneymanagerex/moneymanagerex/issues/3853#issuecomment-942636569) you can find detailed instructions. With Fedora 34, you'll need to download and compile wxwidgets, but with fedora 35, you can use the native ones.

**Comment by Zeit08** [comment]
Hi, I recently compiled from source on a Debian based distro, but now I am on Fedora 36 (beta) and indeed the .rpm release package is not launching. The errors are shown here
![image](https://user-images.githubusercontent.com/45097367/163031162-7dbb764a-b163-4d47-b6c7-ab1101b7c639.png).

Can I ask what do I need to to do make the .rpm package work? And if it is not going to work anyway, is there a step by step guide to compile mmex on Fedora 36?

**Comment by jota2016** [comment]
Hi,
In mi previous post there is a link with detailed instructions.
I just compiled v1.5.14 on Fedora 36 Beta with no problem. 
@Zeit08 how did you compile the package? 

Create a dir for the process
`mkdir /opt/git`

Init as a git repository
`cd /opt/git`
`git init`

Clone mmex repo
`git clone --recursive https://github.com/moneymanagerex/moneymanagerex`
-By default, you are in the master branch.
-We want to use v1.5.14, so:
`cd moneymanagerex`
`git checkout v1.5.14`
-Create build directory
`mkdir build1514 && cd build1514`

Compile mmex
`export MAKEFLAGS=-j8`
`cmake -DCMAKE_CXX_FLAGS="-w" -DCMAKE_BUILD_TYPE=Release -DwxWidgets_CONFIG_EXECUTABLE=/usr/libexec/wxGTK/wx-config ..`
`cmake --build . --target package`

**Comment by Zeit08** [comment]
My previous screenshot was from Software app, while trying to install the .rpm package available in release page here on GitHub. A few months ago I compiled mmex on Debian and had no issues, I used [this guide](https://github.com/moneymanagerex/moneymanagerex/issues/2472#issuecomment-959494974). As I saw wxWidgets were no longer needed I thought I need to compile differently, also because I changed from Debian to Fedora. You showed me that compilation process is the same.

I saw your previous linked guide but I was unsure how to modify it removing the wxWidgets part.

But eventually I still downloaded and compiled wxWidgets because the native ones were not found when following your guide here. Also I had many other errors because of missing packages, like _ccache, libcurl-devel, cmake, lsb_release_ and maybe a few others. I **suppose** it is because I am using a beta release (Fedora 36) so in stable version all those libraries and packages will be there without needing to manually look for them and install them.

Moreover, while compiling mmex I got another error: _RPM package requires rpmbuild executable_
So I had to install _rpm-build_ to solve that.

Eventually, rpm was generated, I double clicked it and it installed successfully!
